src/jarabe/model/bundleregistry.py

```diff
@@ -140,6 +140,23 @@ def __file_monitor_changed_cb(self, monitor, one_file, other_file,
             self.add_bundle(one_file.get_path(), set_favorite=True)
         elif event_type == Gio.FileMonitorEvent.DELETED:
             self.remove_bundle(one_file.get_path())
+            for root in GLib.get_system_data_dirs():
+                root = os.path.join(root, 'sugar', 'activities')
+
+                try:
+                    dir_list = os.listdir(root)
+                except OSError:
+                    logging.debug('Can not find GLib system dir %s', root)
+                    continue
+                activity_dir = os.path.basename(one_file.get_path())
+                try:
+                    bundle = bundle_from_dir(os.path.join(root, activity_dir))
+                except MalformedBundleException:
+                    continue
+
+                path = bundle.get_path()
```

Comment:
I'm getting errors for when `bundle is None`.  Maybe add a check for that in there?

I also now understand why you didn't keep the logic the same between the 2 functions.  My bad for making that comment before.